### PR TITLE
xtest_1000: Make fake and corrupt TAs readable by others

### PR DESCRIPTION
**Pull request by vchong**
Signed-off-by: Victor Chong <victor.chong@linaro.org>

### CODE REVIEW

**Comment by vchong** [comment]
If for any (stupid?) reason xtest is run with some higher permission that tee-supp doesn't have, then supp will not have permission to access the fake and corrupted TAs so 1008.4 and 1008.5 will fail. This patch will avoid this problem, but admittedly, this is probably a very rare case. If we deem that this is too rare and can be ignored, then we can just abandon this patch.

File: host/xtest/xtest_1000.c [hunk 1]
```diff
@@ -440,9 +441,19 @@ static FILE *open_ta_file(const TEEC_UUID *uuid, const char *mode,
 			  bool for_write)
 {
 	char buf[PATH_MAX];
+	FILE *ret;
 
 	uuid_to_full_name(buf, sizeof(buf), uuid, for_write);
-	return fopen(buf, mode);
+	ret = fopen(buf, mode);
+	if (ret && for_write)
+		if (chmod(buf, 0666)) {
```
Comment (jforissier) [inline host/xtest/xtest_1000.c]: Why not set mode to 0666 in `fopen()` instead?

File: host/xtest/xtest_1000.c [hunk 2]
```diff
@@ -440,9 +441,19 @@ static FILE *open_ta_file(const TEEC_UUID *uuid, const char *mode,
...
+			fprintf(stderr, "chmod(\"%s\"): %s",
+					buf, strerror(errno));
+			printf("chmod error! This test will fail if " \
+				"tee-supplicant does not have permission " \
+				"to load this TA file!\n");
```
Comment (jforissier) [inline host/xtest/xtest_1000.c]: You're mentioning read permission by `tee-supplicant` but the subject of the commit is "Make xxx writable". I don't understand.

**Comment by jforissier** [comment]
In my opinion `xtest` should not try to fix an invalid setup. There are so many other things that can be setup wrong. Just report errors in a useful manner as they occur and it's enough.

**Comment by vchong** [comment]
> What's the problem? Why does this change fix it?

If say xtest is run as root, it will create the fake and corrupted TA with mode 0600. If tee-supplicant is not root, then it will not be able to load the TAs and fail.

> In my opinion xtest should not try to fix an invalid setup. There are so many other things that can be setup wrong. Just report errors in a useful manner as they occur and it's enough.

Yes, if we deem running xtest as root and tee-supplicant as non-root as an invalid setup, then we can just abandon this.

File: host/xtest/xtest_1000.c (hunk 1 above)
Comment (vchong) [inline host/xtest/xtest_1000.c]: That was my first thought too but looking it up, it seems like `fopen()` can only do the `r, w, r+`, etc stuffs?

File: host/xtest/xtest_1000.c (hunk 2 above)
Comment (vchong) [inline host/xtest/xtest_1000.c]: You're right. Strictly speaking we only need 0644, not 0666. Was maybe overdoing the future-proofing. Can change `writable` to `readable` too if necessary.

**Comment by jenswi-linaro** [comment]
It sounds like umask is wrong for xtest, with a normal umask of 022 it should create file that are 0644.

**Comment by jenswi-linaro** [comment]
Before you start xtest at the prompt, what does umask say?

**Comment by vchong** [comment]
> Before you start xtest at the prompt, what does umask say?

077

**Comment by jenswi-linaro** [comment]
There's the problem. I'm not that happy about to hard code an umask 022 in xtest since it's a matter of the environment. Perhaps it's acceptable to just update the instructions for how xtest should be launched in Android.

**Comment by vchong** [comment]
Thanks! Sounds like something like `umask 22 && xtest && umask 77` would work.

**Comment by jenswi-linaro** [comment]
Or even `(umask 022 && xtest)`, the `()` are important.

**Pull request closed without merging**